### PR TITLE
perf: Batch Plain Parquet UTF-8 verification

### PR DESCRIPTION
**Pull request by coastalwhite**
```
$ POLARS_MAX_THREADS=1 poop './plparbench-before 10 plain.parquet auto' './plparbench-after 10 plain.parquet auto'

Benchmark 1 (3 runs): ./plparbench-before 10 plain.parquet auto
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          13.0s  ±  701ms    12.5s  … 13.8s           0 ( 0%)        0%
  peak_rss           1.46GB ±  407KB    1.46GB … 1.46GB          0 ( 0%)        0%
  cpu_cycles         37.9G  ± 84.2M     37.8G  … 38.0G           0 ( 0%)        0%
  instructions        113G  ± 46.7M      113G  …  113G           0 ( 0%)        0%
  cache_references    426M  ±  397K      425M  …  426M           0 ( 0%)        0%
  cache_misses        111M  ±  777K      110M  …  112M           0 ( 0%)        0%
  branch_misses       634K  ± 53.6K      597K  …  695K           0 ( 0%)        0%
Benchmark 2 (3 runs): ./plparbench-after 10 plain.parquet auto
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          8.55s  ± 36.1ms    8.51s  … 8.58s           0 ( 0%)        ⚡- 34.2% ±  8.7%
  peak_rss           1.46GB ±  179KB    1.46GB … 1.46GB          0 ( 0%)          -  0.0% ±  0.0%
  cpu_cycles         20.6G  ± 3.93M     20.6G  … 20.6G           0 ( 0%)        ⚡- 45.6% ±  0.4%
  instructions       50.0G  ± 11.1M     50.0G  … 50.0G           0 ( 0%)        ⚡- 55.6% ±  0.1%
  cache_references    377M  ±  299K      377M  …  378M           0 ( 0%)        ⚡- 11.4% ±  0.2%
  cache_misses        102M  ± 94.0K      102M  …  102M           0 ( 0%)        ⚡-  8.1% ±  1.1%
  branch_misses       237K  ± 8.80K      229K  …  247K           0 ( 0%)        ⚡
```

For `PLAIN` string page decoding, this can significantly improve the performance.

### CODE REVIEW

File: crates/polars-parquet/src/arrow/read/deserialize/binview.rs [hunk 1]
```diff
@@ -626,11 +635,20 @@ impl utils::Decoder for BinViewDecoder {
...
+
         if self.check_utf8.load(Ordering::Relaxed) {
-            // @TODO: Better error message
-            values
-                .validate_utf8(buffer_offset, views_offset)
-                .map_err(|_| ParquetError::oos("Binary view contained invalid UTF-8"))?
+            // This is a small trick that allows us to check the Parquet buffer instead of the view
+            // buffer. Batching the UTF-8 verification is more performant. For this to be allowed,
+            // all the interleaved lengths need to be valid UTF-8.
+            if max_length < 128 {
```
Comment (ritchie46) [inline crates/polars-parquet/src/arrow/read/deserialize/binview.rs:644]: Does this validate all strings concatenated? If so, I am not sure this is valid.

Comment (coastalwhite) [reply crates/polars-parquet/src/arrow/read/deserialize/binview.rs:644]: Yes, it does. I am reasonably certain it is valid. Why wouldn't it be valid?

Comment (ritchie46) [reply crates/polars-parquet/src/arrow/read/deserialize/binview.rs:644]: for instance **pi** contains of two unicode chars.

[206 160] is valid pi.

But `[206 | 160]` with a word boundary after 206 is invalid utf8.

Comment (coastalwhite) [reply crates/polars-parquet/src/arrow/read/deserialize/binview.rs:644]: We check that all the lengths are valid UTF-8 (`max_length < 128`). The first byte of a length is then always `0`, which would always follow any individual string. In UTF-8, `0` is never a valid byte beyond the first byte.

Comment (ritchie46) [reply crates/polars-parquet/src/arrow/read/deserialize/binview.rs:644]: Resolved async. :+1:

Comment (orlp) [reply crates/polars-parquet/src/arrow/read/deserialize/binview.rs:644]: I believe that 'first byte of a length' should be 'top bit of the first byte of the length', and in UTF-8 multi-byte characters can't have the top bit set to 0.

**Pull request merged**